lib/rules/no-loss-of-precision.js

```diff
@@ -0,0 +1,196 @@
+         * @returns {Object} the object containing the the integer's coefficient and order of magnitude
+         */
+        function normalizeFloat(stringFloat) {
+            if (stringFloat.startsWith("0") || stringFloat.startsWith(".")) {
+                const decimalDigits = stringFloat.split(".").pop();
+                const significantDigits = removeLeadingZeros(decimalDigits);
+
+                return {
+                    magnitude: significantDigits.length - decimalDigits.length - 1,
+                    coefficient: addDecimalPointToNumber(significantDigits)
+                };
+
+            }
+            return {
+                magnitude: stringFloat.indexOf(".") - 1,
+                coefficient: addDecimalPointToNumber(stringFloat.replace(".", ""))
+
+            };
+        }
+
+
+        /**
+         * Converts a base ten number to proper scientific notation
+         * @param {string} stringNumber the string representation of the base ten number to be converted
+         * @returns {string} the number converted to scientific notation
+         */
+        function convertNumberToScientificNotation(stringNumber) {
+            const absoluteValue = stringNumber.startsWith("-") ? stringNumber.replace("-", "") : stringNumber;
+            const splitNumber = absoluteValue.replace("E", "e").split("e");
+            const originalCoefficient = splitNumber[0];
+            const normalizedNumber = stringNumber.includes(".") ? normalizeFloat(originalCoefficient)
+                : normalizeInteger(originalCoefficient);
+            const normalizedCoefficient = normalizedNumber.coefficient;
+            const magnitude = splitNumber.length > 1 ? (parseInt(splitNumber[1], 10) + normalizedNumber.magnitude)
+                : normalizedNumber.magnitude;
+
+            return `${normalizedCoefficient}e${magnitude}`;
+
+        }
+
+        /**
+         * Checks that the user-intended base ten number equals the actual number after is has been converted to the Number type
+         * @param {Node} node the node being evaluated
+         * @returns {boolean} true if they don not match
+         */
+        function checkBaseTenNumber(node) {
+            const normalizedRawNumber = convertNumberToScientificNotation(node.raw);
+            const requestedPrecision = normalizedRawNumber.split("e")[0].replace(".", "").length;
+
+            if (requestedPrecision > 100) {
+                return true;
+            }
+            const storedNumber = node.value.toPrecision(requestedPrecision);
+            const normalizedStoredNumber = convertNumberToScientificNotation(storedNumber);
+
+            return normalizedRawNumber !== normalizedStoredNumber;
+        }
+
+
+        /**
+         * Checks that the user-intended number equals the actual number after is has been converted to the Number type
+         * @param {Node} node the node being evaluated
+         * @returns {boolean} true if they match
```

Comment:
Is this comment accurate? It seems like `checkNumber` returns true if the mathematical value and rounded value do *not* match.

(It might also be a good idea to rename these function to something like `losesPrecision`, `baseTenNumberLosesPrecision`, etc. to make the behavior clearer.)